src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -981,12 +987,11 @@ internal TdsOperationStatus TryProcessHeader()
             else
             {
                 // normal header processing...
-                _messageStatus = _inBuff[_inBytesUsed + 1];
-                _inBytesPacket = (_inBuff[_inBytesUsed + TdsEnums.HEADER_LEN_FIELD_OFFSET] << 8 |
-                                              _inBuff[_inBytesUsed + TdsEnums.HEADER_LEN_FIELD_OFFSET + 1]) - _inputHeaderLen;
-                _spid = _inBuff[_inBytesUsed + TdsEnums.SPID_OFFSET] << 8 |
-                                              _inBuff[_inBytesUsed + TdsEnums.SPID_OFFSET + 1];
-#if NET
+                ReadOnlySpan<byte> header = _inBuff.AsSpan(_inBytesUsed, TdsEnums.HEADER_LEN);
+                _messageStatus = Packet.GetStatusFromHeader(header);
+                _inBytesPacket = Packet.GetDataLengthFromHeader(header);
+                _spid = Packet.GetSpidFromHeader(header);
+#if NET6_0_OR_GREATER
```

Comment:
I believe `NET` is the new define, right?
```suggestion
#if NET
```